policy_agent/policy_agent.py

```diff
@@ -0,0 +1,655 @@
+        return "%s/%s" % (self.namespace, self.name)
+
+    def apply_ns_policy(self):
+        namespace_profile = "namespace_%s" % self.namespace
+
+        if self.ep_id and _datastore_client.profile_exists(namespace_profile) and _datastore_client.get_endpoints(endpoint_id=self.ep_id):
+            try:
+                _log.info("Applying %s NS policy to EP %s" % \
+                          (self.namespace, self.ep_id))
+                _datastore_client.append_profiles_to_endpoint(profile_names=[namespace_profile],
+                                                              endpoint_id=self.ep_id)
+            except ProfileAlreadyInEndpoint:
+                _log.warning("Apply %s to Pod %s : Profile Already exists" % \
+                             (namespace_profile, self.key))
+                pass
+            return True
+        else:
+            _log.error("Pod Resource %s found before Namespace Resource %s" % \
+                       (self.name, self.namespace))
+            return False
+
+    def remove_default_profile(self):
+        """
+        Remove the default reject all rule programmed by the plugin.
+        """
+        default_profile = DEFAULT_PROFILE_REJECT
+        try:
+            _log.info("Removing Default Profile")
+            _datastore_client.remove_profiles_from_endpoint(profile_names=[default_profile],
+                                                            endpoint_id=self.ep_id)
+            return True
+        except ProfileNotInEndpoint:
+            _log.info("Default Profile not on Endpoint")
+            return True
+        else:
+            _log.info("Default Profile Removal Failed")
+            return False
+
+
+class Endpoints(Resource):
+
+    def init_from_json(self, json):
+        self.kind = KIND_ENDPOINTS
+        self.name = json["metadata"]["name"]
+        self.namespace = json["metadata"]["namespace"]
+        self.subsets = json["subsets"]
+
+    @property
+    def key(self):
+        return "%s/%s" % (self.namespace, self.name)
+
+    def generate_svc_profiles_pods(self):
+        """
+        Endpoints objects contain a list of subsets
+        containing pod and policy info
+        subsets: [
+            {
+                addresses: [
+                    ...pods...
+                ],
+                ports: [
+                    ...associated ports...
+                ]
+            },
+            {
+                addresses: [
+                    ...pods...
+                ],
+                ports: [
+                    ...associated ports...
+                ]
+            },
+        ]
+        :return: A generated dict of profile-pod associations
+        :rtype: a dict of profiles mapping to a list of assoc pods
+        """
+        def verify_args(ports_obj, try_arg):
+            try:
+                arg = ports_obj[try_arg]
+            except KeyError:
+                arg = None
+            return arg
+
+        self.association_list = {}
```

Comment:
I need it for reference when deleting profiles. Although I suppose it makes returning redundant